src/status_im/contexts/wallet/wallet_connect/processing_events.cljs

```diff
@@ -1,42 +1,48 @@
 (ns status-im.contexts.wallet.wallet-connect.processing-events
-  (:require [clojure.string :as string]
+  (:require [cljs-bean.core :as bean]
+            [clojure.string :as string]
             [native-module.core :as native-module]
             [re-frame.core :as rf]
             [status-im.constants :as constants]
             [status-im.contexts.wallet.wallet-connect.core :as wallet-connect-core]
+            [status-im.contexts.wallet.wallet-connect.transactions :as transactions]
             [taoensso.timbre :as log]
             [utils.transforms :as transforms]))
 
 (rf/reg-event-fx
- :wallet-connect/process-session-request
- (fn [{:keys [db]} [event]]
-   (let [method (wallet-connect-core/get-request-method event)
+ :wallet-connect/show-request-modal
+ (fn [{:keys [db]}]
+   (let [event  (get-in db [:wallet-connect/current-request :event])
+         method (wallet-connect-core/get-request-method event)
          screen (wallet-connect-core/method-to-screen method)]
      (if screen
-       {:db (assoc-in db [:wallet-connect/current-request :event] event)
-        :fx [(condp = method
-               constants/wallet-connect-eth-send-transaction-method
-               [:dispatch [:wallet-connect/process-eth-send-transaction]]
-
-               constants/wallet-connect-eth-sign-method
-               [:dispatch [:wallet-connect/process-eth-sign]]
+       {:fx [[:dispatch [:open-modal screen]]]}
+       (log/error "Didn't find screen for Wallet Connect method"
```

Comment:
Do you think users will never hit this branch of the code? If there's a minor chance it's better to show them a message on the UI with some indication.

I think this strategy can be better sometimes than asking them to share logs, because logs violate users' privacy in many aspects IMO. I personally don't like to share local logs from any application I use.

More like raising awareness @clauxx in case it gives you ideas.

Reply:
It's pretty unlikely it will happen, as we agree with the dapp on which RPC methods (and by extention screens) we support when we establish the connection. If we get a request with a method we don't support, we shouldn't show anything to the user IMO. The check is more to make sure we don't hit errors during development. 

I fully agree that we should have better ways to report errors without sharing all the logs via email though :D

Reply:
Cool, even better if we don't need to show errors to users. Thanks for the explanation.